packages/ts-codegen/src/react-query.ts

```diff
@@ -6,53 +6,77 @@ import * as w from 'wasm-ast-types';
 import * as t from '@babel/types';
 import { writeFileSync } from 'fs';
 import generate from "@babel/generator";
-import { findAndParseTypes, findQueryMsg } from "./utils";
-import { ReactQueryOptions } from "wasm-ast-types";
+import { findAndParseTypes, findExecuteMsg, findQueryMsg } from "./utils";
+import { getMessageProperties, ReactQueryOptions } from "wasm-ast-types";
 
 
 
-export default async (name: string, schemas: any[], outPath: string, options?: ReactQueryOptions) => {
+export default async (contractName: string, schemas: any[], outPath: string, options?: ReactQueryOptions) => {
 
-    const ReactQueryFile = pascal(`${name}Contract`) + '.react-query.ts';
-    const Contract = pascal(`${name}Contract`)
+    const ReactQueryFile = pascal(`${contractName}Contract`) + '.react-query.ts';
+    const Contract = pascal(`${contractName}Contract`)
 
     const QueryMsg = findQueryMsg(schemas);
+    const ExecuteMsg = findExecuteMsg(schemas);
     const typeHash = await findAndParseTypes(schemas);
 
-    let QueryClient = null;
-    let ReadOnlyInstance = null;
+    const ExecuteClient = pascal(`${contractName}Client`);
+    const QueryClient = pascal(`${contractName}QueryClient`);
 
     const body = [];
 
-    body.push(
-        w.importStmt(['useQuery', 'UseQueryOptions'], 'react-query')
-    );
+    const reactQueryImports = ['useQuery', 'UseQueryOptions']
+    const clientImports = []
+
+    QueryMsg && clientImports.push(QueryClient)
 
+    // check that there are commands within the exec msg
+    const shouldGenerateMutationHooks = ExecuteMsg && options?.mutations && options?.v4 && getMessageProperties(ExecuteMsg).length > 0
+
+    if (shouldGenerateMutationHooks) {
+        body.push(w.importStmt(['ExecuteResult'], '@cosmjs/cosmwasm-stargate'));
+        body.push(w.importStmt(['Coin', 'StdFee'], '@cosmjs/amino'))
+        reactQueryImports.push(...['UseMutationOptions', 'useMutation'])
```

Comment:
do you mean `[].push.apply(reactQueryImports, ['UseMutationOptions', 'useMutation'])` ?

Reply:
Afaik this works because the signature is `push(...items: T[])` ([docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push)) and I've verified a generated output to be:
```ts
import { useQuery, UseQueryOptions, UseMutationOptions, useMutation } from "@tanstack/react-query";
```

Reply:
Though with that being said I suppose it could read:
```ts
reactQueryImports.push('UseMutationOptions', 'useMutation')
```